### ISSUE TITLE
Add support for GridX

### ISSUE THREAD

**Issue by aerwin** [maqetta/maqetta#2034]
This is a general placeholder for adding GridX support. More design details to follow.

**Comment by aerwin** [comment]
This will be dependent on #1811.

**Comment by aerwin** [comment]
With the changes under de7b37c85d5fb4cdab85eec729337beea9b078d2 that I just merged into master there is a new gridx plugin that adds a GridX control to the Dojo container in the widget palette that can be dropped onto HTML pages. There is a fair amount of base functionality present in terms of being able to use Smart Input to change the contents of an instance of GridX. But, there are still some open clean-up issues that I will be looking into:
- GridX widget will only work when running from one of the launchers (e.g., maqetta-single-user) in Eclipse. When run from a build, if you drop GridX onto a page you'll see some errors in the console that it cannot find some modules in Dojo 1.7 plugin. This is being worked in a separate issues (#2067)
- When using Smart Input to change GridX to point at a file-based data store, data does not immediately appear in the GridX widget. You must bring up Smart Input and hit OK again, show source for the page, or re-open the file.
- Deleting GridX from a page fails.
- List of GridX attributes in Properties palette is incomplete.
- Perhaps others that I'll find as a test GridX and DataGrid harder.

Aside from the above, I had to make some changes to GridX source -- most notably, related to setting of cacheClass that doesn't work well declaratively without my modifications. I need to talk to GridX team about those changes to see if they can update their code or if they can give me an alternative workaround.
